src/codeu/chat/client/core/Context.java

```diff
@@ -33,6 +34,10 @@ public Context(ConnectionSource source) {
     this.controller = new Controller(source);
   }
 
+  public ServerInfo getInfo() {
+    return ((View) view).getInfo();
```

Comment:
Whoops. I meant the declaration `private final BasicView view` can be changed to `private final View view,` since we're using it as a `View` only in this class.